test/node-rpc-test.js

```diff
@@ -5,59 +5,512 @@
+      const error0hex = 'Param #0 must be a hex string.';
+      const error1hex = 'Param #1 must be a hex string.';
+
+      const tests = [
+        {args: [[]], message: 'Invalid TXIDs.', code: -8},
+        // help
+        {args: [], message: /^gettxoutproof/, code: -1},
+        {args: ['test'], message: error0array, code: -3},
+        {args: [['test']], message: error0hex, code: -3},
+        {args: [['00ff']], message: error0hex, code: -3},
+        {args: [[hash, '00ff']], message: error1hex, code: -3},
+        {args: [[hash], '0f00'], message: error1hex, code: -3},
+        {args: [[hash, hash]], message: /duplicate/i, code: -8}
+      ];
+
+      for (const test of tests) {
+        // @see https://github.com/bcoin-org/bcurl/pull/16
+        const code = test.code >>> 0;
+
+        await assert.rejects(async () => {
+          await nclient.execute('gettxoutproof', test.args);
+        }, {
+          type: 'RPCError',
+          message: test.message,
+          code: code
+        });
+      }
+    });
+
+    it('should fail with wrong block', async () => {
+      const hash = '00'.repeat(32);
+
+      await assert.rejects(async () => {
+        await nclient.execute('gettxoutproof', [[hash], hash]);
+      }, {
+        type: 'RPCError',
+        message: 'Block not found.',
+        code: -1 >>> 0
+      });
+    });
+
+    it('should fail for mempool tx', async () => {
+      await assert.rejects(async () => {
+        await nclient.execute('gettxoutproof', [[mempoolTXID]]);
+      }, {
+        type: 'RPCError',
+        message: 'Transaction not yet in block.',
+        code: -5 >>> 0
+      });
+    });
+
+    it('should fail for spent coin w/o blockhash and txindex', async () => {
+      // block 0 has spent txid
+      const txid = txids[0][0];
+
+      await assert.rejects(async () => {
+        await nclient.execute('gettxoutproof', [[txid]]);
+      }, {
+        type: 'RPCError',
+        message: 'Block not found.',
+        code: -1 >>> 0
+      });
+    });
+
+    it('should fail getting proof from two blocks', async () => {
+      // different blocks
+      const list = [
+        txids[2][0],
+        txids[3][0]
+      ];
+
+      await assert.rejects(async () => {
+        await nclient.execute('gettxoutproof', [list]);
+      }, {
+        type: 'RPCError',
+        message: 'Block does not contain all txids.',
+        code: -25 >>> 0
+      });
+    });
+
+    it('should fail getting proof from wrong block', async () => {
+      // txid in one block, blockhash of another.
+      const txid = txids[2][0];
+      const blockhash = blockhashes[3];
+
+      await assert.rejects(async () => {
+        await nclient.execute('gettxoutproof', [[txid], blockhash]);
+      }, {
+        type: 'RPCError',
+        message: 'Block does not contain all txids.',
+        code: -25 >>> 0
+      });
+    });
+
+    it('should verify in spv mode and pruned mode', async () => {
```

Comment:
Would be cool to see an invalid or malleated proof fail as well.

Reply:
done.